src/univariate/discrete/discretenonparametric.jl

```diff
@@ -156,8 +156,10 @@ end
 
 minimum(d::DiscreteNonParametric) = first(support(d))
 maximum(d::DiscreteNonParametric) = last(support(d))
-insupport(d::DiscreteNonParametric, x::Real) =
+insupport(d::DiscreteNonParametric{T}, x::T) where {T} =
     length(searchsorted(support(d), x)) > 0
+insupport(d::DiscreteNonParametric{T}, x::Number) where {T<:Number} =
+    length(searchsorted(support(d), convert(T, x))) > 0
```

Comment:
I would prefer `support` to do the right thing here and not `insupport` to second guess the needed transformation

Reply:
I'm not sure what you expect `support` to do here? It's just returning the support, and then we're checking for presence of the value in that set.